beacon-chain/state/field_trie.go

```diff
@@ -213,17 +219,22 @@ func handleEth1DataSlice(val []*ethpb.Eth1Data, indices []uint64, convertAll boo
 	}
 	if convertAll {
 		for i := range val {
-			err := rootCreater(val[i])
+			err := rootCreator(val[i])
 			if err != nil {
 				return nil, err
 			}
 		}
 		return roots, nil
 	}
-	for _, idx := range indices {
-		err := rootCreater(val[idx])
-		if err != nil {
-			return nil, err
+	if len(val) > 0 {
+		for _, idx := range indices {
+			if idx > uint64(len(val))-1 {
+				return nil, fmt.Errorf("index %d greater than number of items in eth1 data slice %d", idx, len(val))
+			}
+			err := rootCreator(val[idx])
+			if err != nil {
+				return nil, err
+			}
```

Comment:
This looks like duplicated code from handleByteArrays. Can you abstract this logic to a common method?